### PR TITLE
Improve documents for the `--font-path` arg & `TYPST_FONT_PATHS` env

### LINKED ISSUE #1708: Improved TYPST_FONT_PATHS documentation
_Originally posted by @luochen1990 in https://github.com/typst/typst/issues/649#issuecomment-1507317901_:

> Hoping for a more detailed description about this ENV variable,  e.g.   WHAT is the expected seperator when there is more than one path;   Which directory level should this PATH point to,  'xxx/share/fonts/opentype/SourceCodePro-Black.otf'  or  'xxx/share/fonts/opentype'  or 'xxx/share/fonts' ?

I found I had the same question when trying to resolve font issues. The online documentation only mentions this env var in a brief CHANGELOG note.

### PR DESCRIPTION
**Pull request by Coekjan**
Closes #1708.

### CODE REVIEW

File: crates/typst-cli/src/args.rs [hunk 1]
```diff
@@ -386,6 +386,12 @@ pub struct FontsCommand {
 #[derive(Debug, Clone, Parser)]
 pub struct FontArgs {
     /// Adds additional directories to search for fonts
+    ///
+    /// The environment variable `TYPST_FONT_PATHS` is used as a fallback if
```
Comment (laurmaedje) [inline crates/typst-cli/src/args.rs:390]: Right now it says 
```
      --font-path <DIR>
          Adds additional directories to search for fonts
          
          [env: TYPST_FONT_PATHS=]
```
so this feels redundant. Especially since the same would apply to TYPST_ROOT otherwise, so it would be inconsistent with this change set.

The separator thing is still relevant though.

File: crates/typst-cli/src/args.rs [hunk 2]
```diff
@@ -386,6 +386,12 @@ pub struct FontsCommand {
...
+    /// `--font-path` is not specified.
+    ///
+    /// Paths specified via `--font-path` and `TYPST_FONT_PATHS` are separated by
+    /// the system's path separator (`:` on Unix-like systems and `;` on Windows).
```
Comment (laurmaedje) [inline crates/typst-cli/src/args.rs:394]: I would reword a bit to something like "If multiple paths are specified, they are separated by ...". I think we can skip the "via `--font-path` and `TYPST_FONT_PATHS`".

File: crates/typst-cli/src/args.rs [hunk 3]
```diff
@@ -386,6 +386,12 @@ pub struct FontsCommand {
 #[derive(Debug, Clone, Parser)]
 pub struct FontArgs {
     /// Adds additional directories to search for fonts
```
Comment (laurmaedje) [inline crates/typst-cli/src/args.rs:388]:
```suggestion
    /// Adds additional directories that are recursively searched for fonts
```
maybe this will help with the person having this problem:
> Which directory level should this PATH point to, 'xxx/share/fonts/opentype/SourceCodePro-Black.otf' or 'xxx/share/fonts/opentype' or 'xxx/share/fonts' ?

File: crates/typst-cli/src/args.rs (hunk 1 above)
Comment (Coekjan) [reply crates/typst-cli/src/args.rs]: Thanks for your kind review. I have removed this. :heart:

**Comment by laurmaedje** [comment]
Thanks!
